initialize task valid move for model rook

kirbygit opened the pull request:
- initialize tasks
- added brakeman.ignore for codeclimate cross site scripting error

kirbygit commented on app/models/bishop.rb:
```diff
@@ -8,5 +8,21 @@ def icon
+    end
+  end
+  
+  def vertical_move?(to_x)
```
your paramater is to_x yet you use to_y as condition?

kirbygit commented on app/models/bishop.rb:
```diff
@@ -8,5 +8,21 @@ def icon
+  end
+  
+  def vertical_move?(to_x)
+    if to_y != starting_point_x
```
on this part

Phatkid98 replied:
I must've written the test in both files. It was a late night, lol. I deleted it.

kirbygit commented:
why is bishop.rb included in the commit? is bishop a different piece right? the movement of bishop is diagaonal only not horizontal/vertical

kirbygit commented:
now you deleted the file bishop_spec.rb. Are you aware you just deleted the test for bishop?

nnadkarny commented:
Bishop spec is still on the branch.  
![screenclip](https://user-images.githubusercontent.com/25439324/26999034-72a92df4-4d5e-11e7-8cce-0f74b27e4d0d.png)

kirbygit commented:
not showing on mine might be a bug
<img width="1280" alt="screen shot 2017-06-09 at 7 01 15 pm" src="https://user-images.githubusercontent.com/20747057/26999086-5ffa1c1c-4d46-11e7-976a-2b1615966c6e.png">
<img width="1280" alt="screen shot 2017-06-09 at 7 02 52 pm" src="https://user-images.githubusercontent.com/20747057/26999087-62e77730-4d46-11e7-9ad0-70902a78685b.png">

kirbygit commented:
now its showing on my production repo when I fetched on source tree
<img width="1280" alt="screen shot 2017-06-09 at 7 07 46 pm" src="https://user-images.githubusercontent.com/20747057/26999122-291d8912-4d47-11e7-9515-320575e1435c.png">
<img width="1280" alt="screen shot 2017-06-09 at 7 08 53 pm" src="https://user-images.githubusercontent.com/20747057/26999123-2a143668-4d47-11e7-9032-981a4ecfe07a.png">
![Uploading Screen Shot 2017-06-09 at 7.07.33 PM.png…]()

Phatkid98 commented:
@kirbygit The Bishop spec is in the branch. Checkout the commit

nnadkarny commented:
@Phatkid98 no worries, we can ignore those.  For spec tests, similarity is unavoidable.

kirbygit commented:
- moved this code to module movements and refactored since if return true otherwise it is returning a nil 
`  def horizontal_move?(to_x, to_y)
    to_x != starting_point_x && starting_point_y == to_y
  end

  def vertical_move?(to_y, to_x)
    to_y != starting_point_y && starting_point_x == to_x
  end
`
- refactored some tests suite
- brakeman.ignore for code climate error "Unescaped model attribute" [refer to XSS on brakeman]
- removed unnecessary include Modules to obstructions since it can only be called in a Class
- pr is good to go after fixing minor issues in code climate

ilyakrasnov commented:
Please mark the codeclimate issues as `wont fix` and we are good to go! 👍

ilyakrasnov commented:
👍
בתאריך יום א׳, 11 ביוני 2017 ב-20:06 מאת nnadkarny <notifications@github.com
>:

> Merged #45 <https://github.com/TheCorrespondingSquares/chess-app/pull/45>.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/TheCorrespondingSquares/chess-app/pull/45#event-1118585623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADmkCVZV6JY3FrpIdfuBnRyKoNZopEHMks5sDB6ggaJpZM4N012S>
> .
>
-- 
--
Ilya Krasnov
LinkedIn <https://www.linkedin.com/in/ikrasnov>
